pulsar/schema.go

```diff
@@ -219,6 +226,107 @@ func (ps *ProtoSchema) GetSchemaInfo() *SchemaInfo {
 	return &ps.SchemaInfo
 }
 
+type ProtoNativeSchema struct {
+	SchemaInfo
+}
+
+func NewProtoNativeSchemaWithMessage(message proto.Message, properties map[string]string) *ProtoNativeSchema {
+	return NewProtoNativeSchema(GetProtoNativeSchemaInfo(message), properties)
+}
+
+func NewProtoNativeSchema(protoNativeSchemaDef string, properties map[string]string) *ProtoNativeSchema {
+	pns := new(ProtoNativeSchema)
+	pns.SchemaInfo.Schema = protoNativeSchemaDef
+	pns.SchemaInfo.Type = ProtoNative
+	pns.SchemaInfo.Properties = properties
+	pns.SchemaInfo.Name = "ProtoNative"
+	return pns
+}
+
+func GetProtoNativeSchemaInfo(message proto.Message) string {
+	fileDesc := message.ProtoReflect().Descriptor().ParentFile()
+	fileProtoMap := make(map[string]*descriptorpb.FileDescriptorProto)
+	GetFileProto(fileDesc, fileProtoMap)
+
+	fileDescList := make([]*descriptorpb.FileDescriptorProto, 0, len(fileProtoMap))
+	for k := range fileProtoMap {
+		value := descriptorpb.FileDescriptorProto{
+			Name:             fileProtoMap[k].Name,
+			Package:          fileProtoMap[k].Package,
+			Dependency:       fileProtoMap[k].Dependency,
+			PublicDependency: fileProtoMap[k].PublicDependency,
+			MessageType:      fileProtoMap[k].MessageType,
+			EnumType:         fileProtoMap[k].EnumType,
+			Service:          fileProtoMap[k].Service,
+			Extension:        fileProtoMap[k].Extension,
+			Options:          fileProtoMap[k].Options,
+			SourceCodeInfo:   fileProtoMap[k].SourceCodeInfo,
+			Syntax:           fileProtoMap[k].Syntax,
+		}
+		fileDescList = append(fileDescList, &value)
```

Comment:
```suggestion
        for _, v := range fileProtoMap {
                fileDescList = append(fileDescList, v)
```

Simplify the code. We don't need to copy the object that the pointer `v` references. Golang will handle the pointer well.

Reply:
fixed